doc/examples/plot_hyperparameter.py

```diff
@@ -0,0 +1,166 @@
+"""
+Simple Parallel Model Selection
+===============================
+
+In this example, we'll demonstrate how to quickly write a hyperparameter
+tuning script that evaluates a set of hyperparameters in parallel.
+
+This script will demonstrate how to use the ``ray.remote`` API for functions
+along with ``ray.wait``.
+
+.. important:: For a production-grade implementation of distributed
+    hyperparameter tuning, use `Tune`_, a scalable hyperparameter
+    tuning library built using Ray's Actor API.
+
+.. _`Tune`: https://ray.readthedocs.io/en/latest/tune.html
+"""
+import os
+import numpy as np
+from filelock import FileLock
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+
+import ray
+
+ray.init()
+
+# The number of sets of random hyperparameters to try.
+num_evaluations = 10
+
+
+# A function for generating random hyperparameters.
+def generate_hyperparameters():
+    return {
+        "learning_rate": 10**np.random.uniform(-5, 1),
+        "batch_size": np.random.randint(1, 100),
+        "momentum": np.random.uniform(0, 1)
+    }
+
+
+def get_data_loaders(batch_size):
+    mnist_transforms = transforms.Compose(
+        [transforms.ToTensor(),
+         transforms.Normalize((0.1307, ), (0.3081, ))])
+
+    # We add FileLock here because multiple workers will want to
```

Comment:
`download=True` will only download if not found.

Reply:
(And it shouldn't have their own file paths because realistically, these datasets will be very large and multiple copies of the same data wouldn't make sense)